app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/presentation/main/MainPresenter.kt

```diff
@@ -0,0 +1,33 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.presentation.main
+
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.MarvelRepository
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.applySchedulers
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.plusAssign
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.smartSubscribe
+import com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common.Presenter
+
+class MainPresenter(val view: MainView) : Presenter() {
+
+    private val repository by MarvelRepository.lazyGet()
```

Comment:
there are plenty sources about types of injections and why constructor injection is the best one.